File: lib/layer2_core/light/e_surface_container_high_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class ESurfaceContainerHighLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl12;
      case 'standard': return cstl12;
      case 'blue': return cbl12;
      case 'green': return cgl12;
      case 'red': return crl12;
      case 'nothing': return cnol12;
      default: return cstl12;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/ESurfaceContainerHighLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESurfaceContainerHighLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl12
            AppTheme.STANDARD -> R.color.cstl12
            AppTheme.BLUE -> R.color.cbl12
            AppTheme.GREEN -> R.color.cgl12
            AppTheme.RED -> R.color.crl12
            AppTheme.NOTHING -> R.color.cnol12
        }
    }
}
